New_Authenticator.ipynb

```diff
@@ -0,0 +1,71 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "from jupyterhub.auth import Authenticator\n",
+    "\n",
+    "from tornado import gen\n",
+    "\n",
+    "import csv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   
+   "outputs": [],
+   "source": [
+    "class MyNewAuthenticator(Authenticator):\n",
+    "    with open('Example.csv') as csvfile:\n",
```

Comment:
We can't read the csv at the class level here. It needs to be in the authenticate method.

Try testing your authenticator lower in the notebook to make sure it works:

```python
auth = MyNewAuthenticator()
auth.authenticate(None, {'username': 'me', 'password': 'something'})
```